libraries/disp3D/helpers/geometryinfo/geometryinfo.h

```diff
@@ -122,48 +122,41 @@ class DISP3DSHARED_EXPORT GeometryInfo
 
     //=========================================================================================================
     /**
-    * @brief scdc                  Calculates surface constrained distances on the mesh that is held by the passed MNEBemSurface
+    * @brief scdc                           Calculates surface constrained distances on a mesh.
     *
-    * @param bemSurface            The surface on which distances should be calculated
-    * @param pVecVertSubset        The subset of IDs for which the distances should be calculated
-    * @param dCancelDist           Distances higher than this are ignored, i.e. set to infinity
+    * @param[in] matVertices               The surface on which distances should be calculated.
+    * @param[in/out] vecNeighborVertices   The neighbor vertex information.
+    * @param[in] pVecVertSubset            The subset of IDs for which the distances should be calculated.
+    * @param[in] dCancelDist               Distances higher than this are ignored, i.e. set to infinity.
     *
-    * @return                      A double matrix. One column represents the distances for one vertex inside of the passed subset
+    * @return                               A double matrix. One column represents the distances for one vertex inside of the passed subset
     */
-    static Eigen::MatrixXd scdc(const MNELIB::MNEBemSurface &bemSurface,
+    static Eigen::MatrixXd scdc(const Eigen::MatrixX3f &matVertices,
+                                const QVector<QVector<int> > &vecNeighborVertices,
                                 QVector<qint32> &pVecVertSubset,
```

Comment:
pVecVertSubset is missing a const.

Reply:
or make a copy if needed.

Reply:
a non const reference is needed because vecVertSubset gets changed if it is empty.